### PR TITLE
ROS2 updates for Dashing

### PR DESCRIPTION
**Pull request by klintan**
Some updates needed to make it run on latest Dashing release. Perhaps worth creating a new branch "dashing-devel" to keep the old crystal compatible code?

Not sure what to do with the"tf_prefix" part since I'm not sure it makes sense anymore in ROS2 (searching for the global tf_prefix, since it's all node specific now)

### CODE REVIEW

**Comment by klintan** [comment]
this will probably also fix https://github.com/ros-drivers/nmea_navsat_driver/issues/71 for dashing.

**Comment by evenator** [comment]
I'm going to fork the `ros2` branch to `crystal-devel`, and `ros2` will continue to be the dev branch for dashing and beyond.

File: src/libnmea_navsat_driver/driver.py [hunk 1]
```diff
@@ -275,12 +275,12 @@ def add_sentence(self, nmea_string, frame_id, timestamp=None):
...
-        frame_id = self.get_parameter('frame_id').value or 'gps'
+        frame_id = self.declare_parameter('frame_id', 'gps').value
         """Add the TF prefix"""
         prefix = ""
-        prefix_param = self.get_parameter('tf_prefix').value
+        prefix_param = self.declare_parameter('tf_prefix', prefix).value
         if prefix_param:
-            prefix = self.get_parameter(prefix_param).value
+            prefix = self.declare_parameter(prefix_param).value
             return "%s/%s" % (prefix, frame_id)
```
Comment (evenator) [inline src/libnmea_navsat_driver/driver.py]: Looking at this, I'm wondering if the TF prefix parameter is implemented correctly in ROS 2.

Shouldn't this code just be

```python
frame_id = self.declare_parameter('frame_id', 'gps').value
prefix = self.declare_parameter('tf_prefix', '').value
if len(prefix):
    return '%s/%s" % (prefix, frame_id)'
return frame_id
```

Comment (klintan) [reply src/libnmea_navsat_driver/driver.py]: Yes that makes more sense, given tf_prefix is also deprecated in TF2.

**Comment by klintan** [comment]
Dropped the ball a bit on this one, but pushed the change requested.

**Pull request merged**